src/deploy-cloudrun.ts

```diff
@@ -0,0 +1,156 @@
+    const credentials = core.getInput('credentials'); // Service account key
+    let projectId = core.getInput('project_id');
+    // Flags
+    const envVars = core.getInput('env_vars'); // String of env vars KEY=VALUE,...
+    const region = core.getInput('region') || 'us-central1';
+
+    let installBeta = false; // Flag for installing gcloud beta components
+    let cmd;
+
+    // Find base command
+    if (metadata) {
+      // Deploy service from metadata
+      if (image || name || envVars) {
+        core.warning(
+          'Metadata YAML provided: ignoring `image`, `service`, and `env_vars` inputs.',
+        );
+      }
+      cmd = [
+        'beta',
+        'run',
+        'services',
+        'replace',
+        metadata,
+        '--platform',
+        'managed',
+        '--region',
+        region,
+      ];
+      installBeta = true;
+    } else {
+      // Deploy service with image specified
+      cmd = [
+        'run',
+        'deploy',
+        name,
+        '--image',
+        image,
+        '--quiet',
+        '--platform',
+        'managed',
+        '--region',
+        region,
+      ];
+    }
+    if (!metadata) {
+      // Set optional flags from inputs
+      if (envVars) cmd.push('--update-env-vars', envVars);
+    }
+
+    // Install gcloud if not already installed.
+    if (!setupGcloud.isInstalled()) {
+      const gcloudVersion = await setupGcloud.getLatestGcloudSDKVersion();
+      await setupGcloud.installGcloudSDK(gcloudVersion);
+    }
+
+    // Authenticate gcloud SDK.
+    if (credentials) await setupGcloud.authenticateGcloudSDK(credentials);
+    const authenticated = await setupGcloud.isAuthenticated();
+    if (!authenticated) {
+      core.setFailed('Error authenticating the Cloud SDK.');
+    }
+
+    // set PROJECT ID
+    if (projectId) {
+      await setupGcloud.setProject(projectId);
+    } else if (credentials) {
+      projectId = await setupGcloud.setProjectWithKey(credentials);
+    } else if (process.env.GCLOUD_PROJECT) {
+      await setupGcloud.setProject(process.env.GCLOUD_PROJECT);
+    }
+    // Fail if no Project Id is provided if not already set.
+    const projectIdSet = await setupGcloud.isProjectIdSet();
+    if (!projectIdSet) core.setFailed('No project Id provided.');
+
+    // Install beta components if needed
+    if (installBeta) await setupGcloud.installComponent('beta');
+
+    const toolCommand = setupGcloud.getToolCommand();
+
+    // Get output of gcloud cmd.
+    let output = '';
+    const stdout = (data: Buffer): void => {
+      output += data.toString();
+    };
+    let errOutput = '';
+    const stderr = (data: Buffer): void => {
+      errOutput += data.toString();
+    };
+
+    const options = {
+      listeners: {
+        stderr,
+        stdout,
+      },
+      silent: true,
+    };
+
+    // Run gcloud cmd.
+    try {
+      await exec.exec(toolCommand, cmd, options);
+      // Set url as output.
+      setUrlOutput(errOutput);
```

Comment:
Does gcloud output url to stderr?

Reply:
Yeah a little odd - integration tests should show this working correctly